### DISCUSSION TITLE
Use phlex with dry-initializer

### DISCUSSION THREAD

**Discussion by me-kirill** [yippee-fun/phlex#553] [Help] [answered]
Hi! I want to use phlex with [dry-initializer](https://dry-rb.org/gems/dry-initializer/3.0/)

I tried this:

```
require 'phlex'
require 'dry-initializer'

class Nav < Phlex::HTML
  extend Dry::Initializer
  param :title

  def template
    h1 { @title }
  end
end

puts Nav.call(title: "Hello") # => <h1></h1>
```

But it doesn't work :(

How can I make it work?

**Comment by joeldrapper** [comment, marked as answer]
It looks like `param` is for positional arguments and `option` is for keyword arguments. So you would either need to use `option` instead of `param` here, or initialize the component with a _positional_ argument `Nav.call("Hello")`.

If you're using Dry Initializer, I recommend patching both the `param` and `option` methods to have `reader: false` by default. Readers are likely to conflict with HTML methods (`title` is a popular one you've actually run into in this example).

Internally, you can use instance variables instead of accessor methods (I think this is best practice anyway), and there’s usually no need to reference component attributes externally. I believe you can patch the methods like this.

```ruby
class BaseComponent < Phlex::HTML
  extend Dry::Initializer

  def self.option(*args, **kwargs, &block)
    super(*args, reader: false, **kwargs, &block)
  end

  def self.param(*args, **kwargs, &block)
    super(*args, reader: false, **kwargs, &block)
  end
end
```

Then you can just inherit from `BaseComponent`.

```ruby
class Nav < BaseComponent
  option :title

  def template
    h1 { @title }
  end
end
```

**Reply by me-kirill** [reply]
Perfect. Thank you Joel!
